generator/integration_tests/CMakeLists.txt

```diff
@@ -44,12 +51,102 @@ target_link_libraries(
+add_library(google_cloud_cpp_generator_golden # cmake-format: sort
+            ${google_cloud_cpp_generator_golden_files})
+
+target_include_directories(
+    google_cloud_cpp_generator_golden
+    PUBLIC ${PROJECT_SOURCE_DIR} ${PROJECT_BINARY_DIR}
+           ${PROJECT_BINARY_DIR}/generator/integration_tests)
+target_link_libraries(
+    google_cloud_cpp_generator_golden
+    PUBLIC google_cloud_cpp_generator_golden_protos
+           google-cloud-cpp::grpc_utils
+           google-cloud-cpp::common
+           google-cloud-cpp::api_client_protos
+           google-cloud-cpp::longrunning_operations_protos
+           absl::str_format
+           protobuf::libprotoc)
+google_cloud_cpp_add_common_options(google_cloud_cpp_generator_golden)
+target_compile_options(google_cloud_cpp_generator_golden
+                       PUBLIC ${GOOGLE_CLOUD_CPP_EXCEPTIONS_FLAG})
+
+# The tests require googletest to be installed. Force CMake to use the config
+# file for googletest (that is, the CMake file installed by googletest itself),
+# because the generic `FindGTest` module does not define the GTest::gmock
+# target, and the target names are also weird.
+find_package(GTest CONFIG REQUIRED)
+
+add_library(
+    google_cloud_cpp_generator_golden_testing # cmake-format: sort
+    tests/mock_golden_kitchen_sink_stub.h tests/mock_golden_thing_admin_stub.h)
+target_link_libraries(
+    google_cloud_cpp_generator_golden_testing
+    PUBLIC google_cloud_cpp_testing google_cloud_cpp_generator_golden
+           GTest::gmock_main GTest::gmock GTest::gtest)
+
+set(google_cloud_cpp_generator_golden_tests
+    # cmake-format: sort
+    tests/golden_kitchen_sink_auth_decorator_test.cc
+    tests/golden_kitchen_sink_client_test.cc
+    tests/golden_kitchen_sink_connection_test.cc
+    tests/golden_kitchen_sink_idempotency_policy_test.cc
+    tests/golden_kitchen_sink_logging_decorator_test.cc
+    tests/golden_kitchen_sink_metadata_decorator_test.cc
+    tests/golden_kitchen_sink_option_defaults_test.cc
+    tests/golden_kitchen_sink_stub_factory_test.cc
+    tests/golden_kitchen_sink_stub_test.cc
+    tests/golden_thing_admin_auth_decorator_test.cc
+    tests/golden_thing_admin_client_test.cc
+    tests/golden_thing_admin_connection_test.cc
+    tests/golden_thing_admin_idempotency_policy_test.cc
+    tests/golden_thing_admin_logging_decorator_test.cc
+    tests/golden_thing_admin_metadata_decorator_test.cc
+    tests/golden_thing_admin_option_defaults_test.cc
+    tests/golden_thing_admin_stub_factory_test.cc
+    tests/golden_thing_admin_stub_test.cc
+    tests/plumbing_test.cc)
+
+# Export the list of unit tests to a .bzl file so we do not need to maintain the
+# list in two places.
+export_list_to_bazel("google_cloud_cpp_generator_golden_tests.bzl"
+                     "google_cloud_cpp_generator_golden_tests" YEAR "2020")
+
+# Create a custom target so we can say "build all the tests"
+add_custom_target(generator-golden-tests)
```

Comment:
hm do we do this anywhere else? is it necessary?

Reply:
Removed.